### PR TITLE
fix: Use AWS::Partition Pseudo Parameter in IAM ARNs 

### PR DESCRIPTION
**Pull request by dougireton**
## Which problem is this PR solving?

In AWS GovCloud, the RDS and CloudWatch Logs templates fail because the RDS and CloudWatch Logs templates contain IAM ARNs hardcoded with the AWS commercial partition (aws) instead of (aws-us-gov).

Example Error Message:

```shell
waiting for CloudFormation Stack (arn:aws-us-gov:cloudformation:us-gov-west-1:<redacted>) create: stack status (ROLLBACK_COMPLETE): The following resource(s) failed to create: [LogSubscription]. Rollback requested by user.
    	
    Resource handler returned message: "Invalid request provided: AWS::Logs::SubscriptionFilter. Could not deliver test message to specified Firehose stream. Check if the given Firehose stream is in ACTIVE state. (Service: CloudWatchLogs, Status Code: 400, Request ID: 123abc)" (RequestToken: 1fc17137-260a-91fe-803f-75e5cce376c2, HandlerErrorCode: InvalidRequest)
```

## Short description of the changes

Swap the hardcoded AWS partition with the [AWS Partition Pseudo Parameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-partition), `AWS::Partition`

1. Allow CloudWatch Logs service in non-default AWS Partitions to assume LogStreamRole

2. For RDS Logs TransformLambdaRole, use correct AWSLambdaBasicExecutionRole ARN for non-default AWS Partitions

## How to verify that this has the expected result

Deploy RDS Logs or CloudWatch Logs templates in AWS GovCloud.

### CODE REVIEW

**Comment by MikeGoldsmith** [comment]
Thanks for opening the PR @dougireton.

It looks like one of the CI checks failed because it's based on an old AWS SDK. I'll look at updating that so we can ensure everything works as expected.

Your change itself looks good 👍🏻

**Comment by MikeGoldsmith** [comment]
Created the following PR to fix the validate workflow:
- https://github.com/honeycombio/cloudformation-integrations/pull/38

**Comment by JamieDanielson** [comment]
Per [my comment over here](https://github.com/honeycombio/cloudformation-integrations/pull/38#issuecomment-2163896558), I'm considering that the checks didn't fail because of the deprecated action (though that should get updated either way), but rather because of a missing value. I'm not positive just yet if it's because of the required access to a secret not being available for a PR from a fork (as mentioned on that comment), or if it's the missing environment variable `${AWS::Partition}` 🤔

**Comment by JamieDanielson** [comment]
>I'm not positive just yet if it's because of the required access to a secret not being available for a PR from a fork (as mentioned on that comment), or if it's the missing environment variable `${AWS::Partition}` 🤔

I'm thinking it's the former, based on my read of the AWS::Partition that is just part of the [`role-to-assume`](https://github.com/aws-actions/configure-aws-credentials/blob/c7680c32522b91cf6f7481e49535e154b93bed6d/README.md#oidc-audience). So this change I think has to be committed by one of our team members. I think in situations like this we've just done a branch commit of the same thing to pass CI but will double check how best to approach.

**Comment by JamieDanielson** [comment]
I've merged the PR for updating the workflow in case it helped, but it looks like the check still failed. This will need a bit more research to understand how to make the `AWS::Partition` option work 🤔

**Comment by JamieDanielson** [comment]
> I've merged the PR for updating the workflow in case it helped, but it looks like the check still failed. This will need a bit more research to understand how to make the `AWS::Partition` option work 🤔

Actually the workflow does seem to resolve one of the errors that didn't recognize this option, but it's still failing. It seems to still run against the forked repo, so I am pretty sure it's the permissions issue. We'll look to create a branch that uses this change and see if that fixes the error.

File: templates/rds-logs.yml [hunk 1]
```diff
@@ -199,7 +199,7 @@ Resources:
             Action:
               - "sts:AssumeRole"
       ManagedPolicyArns:
-        - "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+        - "arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
```
Comment (JamieDanielson) [inline templates/rds-logs.yml:202]:
```suggestion
        - !Sub "arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
```

**Pull request closed without merging**